### PR TITLE
Add `platform.getSubnetOnlyValidator` API

### PR DESCRIPTION
**Pull request by StephenButtolph**
## Why this should be merged

This will allow the wallet to fetch the `DeactivationOwner` based on the `validationID`.

This is needed to resolve the TODO in the P-chain wallet in #3440.

## How this works

- Adds `platform.getSubnetOnlyValidator`

## How this was tested

- [X] Extended E2E test

## Need to be documented in RELEASES.md?

Yes.

### CODE REVIEW

**Review by darioush — APPROVED** [review]

File: vms/platformvm/client.go [hunk 1]
```diff
@@ -326,6 +330,69 @@ func (c *client) GetCurrentValidators(
...
+type SubnetOnlyValidator struct {
+	SubnetID              ids.ID
+	NodeID                ids.NodeID
+	PublicKey             *bls.PublicKey
+	RemainingBalanceOwner *secp256k1fx.OutputOwners
+	DeactivationOwner     *secp256k1fx.OutputOwners
+	StartTime             uint64
+	Weight                uint64
+	MinNonce              uint64
+	// Balance is the remaining amount of AVAX this SoV has to pay the
```
Comment (ARR4N) [inline vms/platformvm/client.go:344]: (nit) "has for paying"

It could just be my complete lack of context but "has to pay" led my brain down the path of "is required to pay [entity]".

File: vms/platformvm/client.go [hunk 2]
```diff
@@ -326,6 +330,69 @@ func (c *client) GetCurrentValidators(
...
+	Balance uint64
+}
+
+func (c *client) GetSubnetOnlyValidator(
+	ctx context.Context,
+	validationID ids.ID,
+	options ...rpc.Option,
+) (SubnetOnlyValidator, uint64, error) {
+	res := &GetSubnetOnlyValidatorReply{}
+	err := c.requester.SendRequest(ctx, "platform.getSubnetOnlyValidator", &GetSubnetOnlyValidatorArgs{
```
Comment (ARR4N) [inline vms/platformvm/client.go:355]: (nit) I find this indentation very hard to parse. Perhaps:

```go
	c.requester.SendRequest(
		ctx, "platform.getSubnetOnlyValidator",
		&GetSubnetOnlyValidatorArgs{
			ValidationID: validationID,
		},
		res, options...,
	)
```

**Review by ARR4N — APPROVED** [review]

File: vms/platformvm/service.go [hunk 3]
```diff
@@ -969,6 +970,91 @@ func (s *Service) GetCurrentValidators(_ *http.Request, args *GetCurrentValidato
...
+	Balance avajson.Uint64 `json:"balance"`
+	// Height is the height of the last accepted block
+	Height avajson.Uint64 `json:"height"`
+}
+
+// GetSubnetOnlyValidator returns the SoV if it exists
+func (s *Service) GetSubnetOnlyValidator(r *http.Request, args *GetSubnetOnlyValidatorArgs, reply *GetSubnetOnlyValidatorReply) error {
+	s.vm.ctx.Log.Debug("API called",
+		zap.String("service", "platform"),
+		zap.String("method", "getSubnetOnlyValidator"),
```
Comment (ceyonur) [inline vms/platformvm/service.go:998]: nit: add validationID arg

File: vms/platformvm/service.go [hunk 4]
```diff
@@ -969,6 +970,91 @@ func (s *Service) GetCurrentValidators(_ *http.Request, args *GetCurrentValidato
...
+	)
+
+	s.vm.ctx.Lock.Lock()
```
Comment (ceyonur) [inline vms/platformvm/service.go:1001]: nit: readlock

File: vms/platformvm/service.go [hunk 5]
```diff
@@ -969,6 +970,91 @@ func (s *Service) GetCurrentValidators(_ *http.Request, args *GetCurrentValidato
...
+	defer s.vm.ctx.Lock.Unlock()
+
+	sov, err := s.vm.state.GetSubnetOnlyValidator(args.ValidationID)
+	if err != nil {
+		return fmt.Errorf("fetching SoV %q failed: %w", args.ValidationID, err)
+	}
+
+	var remainingBalanceOwner message.PChainOwner
+	if _, err := txs.Codec.Unmarshal(sov.RemainingBalanceOwner, &remainingBalanceOwner); err != nil {
```
Comment (ceyonur) [inline vms/platformvm/service.go:1010]: nit: maybe we can add getter methods `sov.GetRemainingBalanceOwner()` and `sov.GetDeactivationOwner()` and move unmarshallings to there?

File: vms/platformvm/service.go [hunk 6]
```diff
@@ -969,6 +970,91 @@ func (s *Service) GetCurrentValidators(_ *http.Request, args *GetCurrentValidato
...
+	// PublicKey is the compressed BLS public key of the validator
+	PublicKey             types.JSONByteSlice `json:"publicKey"`
+	RemainingBalanceOwner platformapi.Owner   `json:"remainingBalanceOwner"`
+	DeactivationOwner     platformapi.Owner   `json:"deactivationOwner"`
+	StartTime             avajson.Uint64      `json:"startTime"`
+	Weight                avajson.Uint64      `json:"weight"`
+	MinNonce              avajson.Uint64      `json:"minNonce"`
+	// Balance is the remaining amount of AVAX this SoV has to pay the
+	// continuous fee, according to the last accepted state.
+	Balance avajson.Uint64 `json:"balance"`
```
Comment (ceyonur) [inline vms/platformvm/service.go:989]: can we add zero balance means an inactive validator?

File: vms/platformvm/service.go (hunk 4 above)
Comment (StephenButtolph) [reply vms/platformvm/service.go:1001]: I'm not going to make this change until we have cleaned up the locking structure of the platformvm. Every API in this file grabs a write lock because the locking model makes no sense currently, and this isn't a performance concern.

File: vms/platformvm/service.go (hunk 5 above)
Comment (StephenButtolph) [reply vms/platformvm/service.go:1010]: I'm planning on making a followup to store the owners in structured form - rather than as bytes.
